### DISCUSSION TITLE
How to load custom activities while running the official Elsa server in docker?

### DISCUSSION THREAD

**Discussion by NightWuYo** [elsa-workflows/elsa-core#4758] [Q&A]
The official docker image is the easiest/quickest way to start a Elsa server.
What if I want to load custom activies please?

Ideally, I can compile/build a class library which contains custom activities, and specify a plugin path for the Esla server, so even running the official docker image, we can load custom activities.

**Comment by sfmskywalker** [comment]
Elsa comes with a feature called the Dropins module. Essentially, it allows you to add NuGet packages to a specific folder while your application is running. Elsa automatically recognises these packages. If these packages include activities, they become immediately available for use in your workflows, without the need to restart your application.

However, we're still in the process of documenting this feature. Moreover, we're planning to expand the ways you can deliver NuGet packages to Elsa. Currently, Elsa checks a local directory, which you can configure, for new packages. But we're looking to add more options, like the ability to upload packages directly to a cloud storage service, such as Azure Storage. This would also enable the application to refresh and incorporate the new packages in real-time.

By enhancing this functionality, our goal is to make Elsa Server a robust, standalone product. This would allow you to set up and manage a workflow server with greater ease, even integrating custom code via NuGet packages, all without the need for building a server from scratch.

**Reply by johnwc** [reply]
What about activities that rely on DI and configurations?

**Reply by pramsperger** [reply]
@sfmskywalker, how do you configure the local directory Elsa checks for new packages?
